resources/tenant-onboarding-managed-ad.yaml

```diff
@@ -33,56 +33,60 @@ Parameters:
     MaxLength: 30
     MinLength: 1
     Type: String
-  ADPasswordParam:
-    ConstraintDescription: please specify a valid Parameter Store name.
-    Description: 'SSM Parameter name to use for AD User password:'
-    MaxLength: '1024'
-    MinLength: '1'
+  TenantId:
+    Description: The GUID for the tenant
     Type: String
 Resources:
+  ADPassword:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Name: !Sub /saas-boost/${Environment}/${TenantId}/ACTIVE_DIRECTORY_PASSWORD
+      GenerateSecretString:
+        PasswordLength: 64
+        ExcludePunctuation: true
   ADDirectory:
     Type: AWS::DirectoryService::MicrosoftAD
     Properties:
       Name: !Sub ${Environment}.${AWS::Region}.saasboost.com
       Edition: !Ref Edition
-      Password: !Sub '{{resolve:ssm-secure:${ADPasswordParam}}}'
-      ShortName: !Sub sb-${Environment}
+      Password: !Sub '{{resolve:secretsmanager:${ADPassword}::::}}'
+      ShortName: 
```

Comment:
Short Name can't exceed 15 characters (NETBIOS limit)
WIth `sb-` and `-tenantId` that only leaves 3 characters for the environment. Suggest changing it to just Tenant-`tenantId` which should always fit.